pkg/jobrunaggregator/jobrunaggregatoranalyzer/pass_fail_test.go

```diff
@@ -0,0 +1,146 @@
+		{
+			// Required Passes for 95th percentile is 6
+			// 5 Natural Passes
+			// Multiplier (6 - 5) == 1
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 2
+			// The Disruption value that == 7 gets flipped to pass
+			name:                 "Test 95th Percentile Fuzzy Pass",
+			disruptions:          []int{5, 5, 5, 6, 6, 7, 9, 9, 9, 9},
+			thresholdPercentile:  95,
+			historicalDisruption: 6,
+			status:               testCasePassed,
+			failedCount:          4,
+			successCount:         6,
+		},
+		{
+			// Required Passes for 95th percentile is 6
+			// 4 Natural Passes
+			// Multiplier (6 - 4) == 2
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 2
+			// The Disruption values that == 7 get flipped to passes
+			name:                 "Test 95th Percentile Multi Fuzzy Pass",
+			disruptions:          []int{5, 5, 5, 6, 7, 7, 8, 8, 8, 86},
+			thresholdPercentile:  95,
+			historicalDisruption: 6,
+			status:               testCasePassed,
+			failedCount:          4,
+			successCount:         6,
+		},
+		{
+			// Required Passes for 95th percentile is 6
+			// 4 Natural Passes
+			// Multiplier (6 - 4) == 2
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 2
+			// The Disruption value that == 7 gets flipped to pass
+			// But the 8s and above do not, so we don't have enough passes
+			name:                 "Test 95th Percentile Multi Fuzzy Fail",
+			disruptions:          []int{5, 5, 5, 6, 7, 8, 8, 9, 9, 86},
+			thresholdPercentile:  95,
+			historicalDisruption: 6,
+			status:               testCaseFailed,
+			failedCount:          5,
+			successCount:         5,
+		},
+		{
+			// Required Passes for 80th percentile is 4
+			// 4 Natural Passes
+			name:                 "Test 80th Percentile Pass",
+			disruptions:          []int{0, 0, 1, 1, 2, 2, 2, 2, 2, 2},
+			thresholdPercentile:  80,
+			historicalDisruption: 1,
+			status:               testCasePassed,
+			failedCount:          6,
+			successCount:         4,
+		},
+		{
+			// Required Passes for 80th percentile is 4
+			// 3 Natural Passes
+			// Multiplier (4-3) == 1
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 1
+			// The Disruption values that == 2 get flipped to passes
+			name:                 "Test 80th Percentile Xtra Fuzzy Pass",
+			disruptions:          []int{0, 0, 1, 2, 2, 6, 3, 4, 5, 8},
+			thresholdPercentile:  80,
+			historicalDisruption: 1,
+			status:               testCasePassed,
+			failedCount:          5,
+			successCount:         5,
+		},
+		{
+			// Required Passes for 80th percentile is 4
+			// 2 Natural Passes
+			// Multiplier (4 - 2) == 2
+			// Fuzz Threshold ((historicalDisruption / 5) + 1) = 1
+			// There are no disruption values that get flipped since (2-1)*2 = 2 and our Fuzz Threshold is 1
+			// But the 8s and above do not so we don't have enough passes
+			name:                 "Test 80th Percentile Fuzzy Fail",
+			disruptions:          []int{0, 0, 2, 2, 2, 2, 2, 2, 2, 2},
+			thresholdPercentile:  80,
```

Comment:
If I'm reading correctly, I think the thresholdPercentile number is not actually relevant to the test, would be one less thing for the next person to look at these tests  to think about.

Reply:
thresholdPercentile is used to set the historical disruption for the test

historicalDisruptionStatistic.percentileByIndex[test.thresholdPercentile] = test.historicalDisruption

and passed into the function to look that data up:

weeklyAverageFromTenDays.checkPercentileDisruption(jobRunIDToAvailabilityResultForBackend, historicalDisruptionStatistic, test.thresholdPercentile)

Reply:
Does using different percentiles change any behavior or could you just hard code say 50 or 95 in the test?

Reply:
The requiredNumberOfPasses changes based on the percentiles

workingPercentage := thresholdPercentile // the percentile is our success percentage
requiredNumberOfPasses := requiredPassesByPassPercentageByNumberOfAttempts[numberOfAttempts][workingPercentage]

Reply:
Ahh I get it now, thanks!